src/boot/efi/efi-string.c

```diff
@@ -272,6 +274,453 @@ bool efi_fnmatch(const char16_t *pattern, const char16_t *haystack) {
+        [NOERR(EFI_ACCESS_DENIED)]        = "Access denied",
+        [NOERR(EFI_NO_RESPONSE)]          = "No response",
+        [NOERR(EFI_NO_MAPPING)]           = "No mapping",
+        [NOERR(EFI_TIMEOUT)]              = "Time out",
+        [NOERR(EFI_NOT_STARTED)]          = "Not started",
+        [NOERR(EFI_ALREADY_STARTED)]      = "Already started",
+        [NOERR(EFI_ABORTED)]              = "Aborted",
+        [NOERR(EFI_ICMP_ERROR)]           = "ICMP error",
+        [NOERR(EFI_TFTP_ERROR)]           = "TFTP error",
+        [NOERR(EFI_PROTOCOL_ERROR)]       = "Protocol error",
+        [NOERR(EFI_INCOMPATIBLE_VERSION)] = "Incompatible version",
+        [NOERR(EFI_SECURITY_VIOLATION)]   = "Security violation",
+        [NOERR(EFI_CRC_ERROR)]            = "CRC error",
+        [NOERR(EFI_END_OF_MEDIA)]         = "End of media",
+        [29]                              = "Reserved (29)",
+        [30]                              = "Reserved (30)",
+        [NOERR(EFI_END_OF_FILE)]          = "End of file",
+        [NOERR(EFI_INVALID_LANGUAGE)]     = "Invalid language",
+        [NOERR(EFI_COMPROMISED_DATA)]     = "Compromised data",
+#  ifdef EFI_HTTP_ERROR
+        [NOERR(EFI_IP_ADDRESS_CONFLICT)]  = "IP address conflict",
+        [NOERR(EFI_HTTP_ERROR)]           = "HTTP error",
+#  endif
+#endif
+};
+
+static const char *process_format_specifier(
+                EFI_STATUS status,
+                const char *format,
+                va_list *ap,
+                char16_t tmp[static SCRATCH_BUF_SIZE],
+                bool *ret_pad_zero,
+                int *ret_precision,
+                int *ret_field_width,
+                const char **ret_str8,
+                const char16_t **ret_str16,
+                const wchar_t **ret_wstr) {
+
+        assert(ap);
+        assert(ret_pad_zero);
+        assert(ret_precision);
+        assert(ret_field_width);
+        assert(ret_str8);
+        assert(ret_str16);
+        assert(ret_wstr);
+
+        bool alternative_form = false, have_field_width = false, long_arg = false;
+        size_t field_length = sizeof(int);
+        int precision = -1, field_width = -1;
+
+        for (;; format++)
+                switch (*format) {
+                case '0':
+                        *ret_pad_zero = true;
+                        break;
+                case '.':
+                        have_field_width = true;
+                        break;
+                case '#':
+                        alternative_form = true;
+                        break;
+
+                case '*':
+                case '1' ... '9': {
+                        int i;
+
+                        if (*format == '*')
+                                i = va_arg(*ap, int);
+                        else {
+                                uint64_t u;
+                                if (!parse_number8(format, &u, &format) || u > INT_MAX)
+                                        assert_not_reached();
```

Comment:
instead of:

```c
if (!x)
        asser_not_reached()
```

write:

```c
assert_se(x);
```

Reply:
This can never happen at runtime, as this is the passed inside of format string branch and will throw a compiler error:
`error: ‘%2147483648i’ directive output of 2147483648 bytes exceeds ‘INT_MAX’`